src/core/resource.cc

```diff
@@ -223,7 +223,18 @@ optional<T> read_setting_V1V2_as(std::string cg1_path, std::string cg2_fname) {
 
 namespace resource {
 
+static
+size_t
+kernel_memory_reservation() {
+    try {
+        return read_first_line_as<size_t>("/proc/sys/vm/min_free_kbytes") * 1024;
+    } catch (...) {
+        return 0;
+    }
+}
+
 size_t calculate_memory(const configuration& c, size_t available_memory, float panic_factor = 1) {
+    available_memory -= kernel_memory_reservation();
```

Comment:
Please, add some logging in case the obtained value doesn't meet the expectations described in the commit message.

Or -- at the end of the `calculate_memory` print info about collected+configured numbers and the evaluated result.

Reply:
Unconditional printing is too noise. I'll print something if the value is unexpected.